2024.07.16 Prefix Sum 개념 학습

choye-sudo opened the pull request:
- codetree IL(lv.4) 01 Simulation의 '겹쳐지지 않는 두 직사각형' 문제 관련해서, 유사 문제를 풀고자 하여 해당 문제를 풀었음
- sliding window : 'window' 영역을 정해서, 한칸씩 이동시키며 window 범위 내의 데이터를 갱신하는 알고리즘
- 해당 문제에서는 범위 내 데이터의 합을 비교하여 최대합을 찾아야 했음.
- sliding window를 배열을 통해 구현함!

choye-sudo committed:
2024.07.16 Prefix Sum 개념 학습

- codetree IL(lv.4) 01 Simulation의 '겹쳐지지 않는 두 직사각형' 문제 관련해서, 유사 문제를 풀고자 함
- slide window 개념을 이해할 수 있게 되었음

llcodingll commented on choye-sudo/prefixSum.java:
```diff
@@ -0,0 +1,37 @@
+        int k = sc.nextInt();
+
+        // 배열을 통해 각 점의 위치를 반영하여 값 저장
+        int[] values = new int[1000001];
```
다음에 해시맵 한 번 써보는 것도 좋을 거 같습니다!

choye-sudo committed:
2024.07.17 : IL 01. 격자안에서 완전탐색 / 양수 직사각형의 최대 크기

- 완전 탐색 문제
- 직사각형을 시작할 왼쪽 상단 모서리를 지정한 후, 해당 모서리에서 가능한 직사각형의 형태를 전부 확인해주었음
- 시간복잡도 : O((nm)^3)임
- 시간복잡도를 줄일 수 있는 방법 : prefix sum을 약간 변형하면 가능함.
1. 지점(왼쪽 상단 모서리)를 (i, j)로 지정
2. 해당 지점으로부터 아래로 쭉 내려가면서 가로가 1인 최대 직사각형의 크기를 열 마다 구해놓기
3. 열을 순회하면서 비교하여 최대 직사각형의 크기를 구하기
